### ISSUE TITLE
Type C screen navigates without interaction

### ISSUE THREAD

**Issue by morcibacsi** [morcibacsi/PSAVanCanBridge#12]
The screen detects menu navigation buttons even if nothing was pressed. This doesn't happen with Chinese OEM displays.
This symptom appears when using the genuine display. 9664644280 and 9663341780 both failed.

In the following video no buttons were pressed.

![IMG_0071](https://user-images.githubusercontent.com/6507173/103208026-fb9a8400-48ff-11eb-848c-2444a466d7f1.gif)

Working display:
![typec_3_chinese](https://user-images.githubusercontent.com/6507173/103209123-91cfa980-4902-11eb-8de2-b730cf4184d0.jpg)

Not working displays:
![typec_1_genuine](https://user-images.githubusercontent.com/6507173/103209141-a01dc580-4902-11eb-9e5a-a5367e5a88e8.jpg)
![typec_2_genuine](https://user-images.githubusercontent.com/6507173/103209143-a0b65c00-4902-11eb-84a3-d1b87b014464.jpg)

**Comment by unobayu** [comment]
i solve this problem by comment line 570 to 631

**Comment by morcibacsi** [comment]
So you mean if you remove the reading from the serial port from PSAVanCanBridgeMain.cpp file, the problem goes away?

So you are reffering to this part

```C
// Line 570:
if (inChar == 'm') {
    PrintVanMessageToSerial = !PrintVanMessageToSerial;
}
if (inChar == 'r') {
    vinFlashStorage->Remove();
}
if (inChar == 'V') {
    serialPort->print("VIN: ");
    for (int i = 0; i < 17; ++i)
    {
        serialPort->write(Vin[i]);
    }
    serialPort->println();
}
if (inChar == 'W')
{
    SendRadioButton(CONST_UP_ARROW);
}

...etc
...etc
...etc
...etc

if (inChar == 'T')
{
    //Serial.println("M pressed");

    for (int i = 0; i < 10; ++i)
    {
        tripInfoHandler->TripButtonPress();
    }
} //this is line 632

```

**Comment by gamelaster** [comment]
Maybe there is some noise in UART, so accidentally the UART commands get triggered?

**Comment by morcibacsi** [comment]
Could be... unfortunately in my setup it doesn't get triggered. Also the user who reported this told me that replacing the screen for another one solves the issue which I don't know how solves the noise issue. Anyways... that code can be removed as it was there for an Android app which was never released, so doesn't have any use now.

**Comment by unobayu** [comment]
yes. i guest reading seial port is the problem. as you know i use my own BT app to control display. so i donot user bt serialport remove that line because its not use. but now the problem for display randomly change is dissapear..

**Comment by morcibacsi** [comment]
So it is not the serial port. Either the display doesn't like dealing with lots of incoming data or something is wrong with the CAN peripheral of the ESP32. There was a similar issue #10 which was resolved by increasing the interval between the trip computer related packets. In 36f50a028f755aa9c7116ae62957ed79133a1b34 I have introduced some dumb "throttling" on writing the CAN bus and it has been reported that the problem is gone.
